sympy/simplify/simplify.py

```diff
@@ -654,43 +677,89 @@ def sum_simplify(s):
                 for j, s_term2 in enumerate(s_t):
                     if not used[j] and i != j:
                         temp = sum_add(s_term1, s_term2, method)
-                        if isinstance(temp, Sum):
+                        if isinstance(temp, Sum) or isinstance(temp, Mul):
                             s_t[i] = temp
                             s_term1 = s_t[i]
                             used[j] = True
 
-    result = Add(*o_t)
+    return used, s_t
 
-    for i, s_term in enumerate(s_t):
-        if not used[i]:
-            result = Add(result, s_term)
+def sum_factor(self, limits=None):
```

Comment:
Sorry, I'm a bit confused by GitHub interface.  This is referring to the function `sum_factor`?
I had called it that because it does sort of the opposite of the `__refactor` method I have in `sum_add`.

In terms of having it work with `expand`, are you referring to this method being called when `expand` is called on Sum?  Or changing the internals of this method such that expand is used.